pkg/client/elasticsearch/query.go

```diff
@@ -114,17 +116,19 @@ func getMetricForPod(
 	defer res.Body.Close()
 
 	if res.IsError() {
-		var e map[string]interface{}
-		if err := json.NewDecoder(res.Body).Decode(&e); err != nil {
-			return timestampedMetric{}, fmt.Errorf("error parsing the response body: %s", err)
-		} else {
-			// Print the response status and error information.
-			return timestampedMetric{}, fmt.Errorf("[%s] %s: %s",
-				res.Status(),
-				e["error"].(map[string]interface{})["type"],
-				e["error"].(map[string]interface{})["reason"],
-			)
+		bodyBytes, err := io.ReadAll(res.Body)
+		if err != nil {
+			return timestampedMetric{}, errors.Wrap(err, "failed to read body")
+		}
+		var errorResponse ErrorResponse
```

Comment:
```suggestion
		var errorResponse estypes.ElasticsearchError
```

Run `go get -u github.com/elastic/go-elasticsearch/v8` to get `github.com/elastic/go-elasticsearch/v8 v8.11.0` and add the following import:

```golang
	estypes "github.com/elastic/go-elasticsearch/v8/typedapi/types"
```